src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -981,140 +982,191 @@ public static void ZipArchive_InvalidVersionToExtract()
                     ZipArchiveEntry secondEntry = updatedArchive.Entries[1];
 
                     Assert.Equal("first.bin", firstEntry.Name);
-                    Assert.Equal(10, firstEntry.Length);
+                    Assert.Equal(s_existingSampleData.Length, firstEntry.Length);
 
                     Assert.Equal("second.bin", secondEntry.Name);
-                    Assert.Equal(4, secondEntry.Length);
+                    Assert.Equal(s_sampleDataToWrite.Length, secondEntry.Length);
 
                     using (Stream entryStream = firstEntry.Open())
                     {
                         byte[] uncompressedBytes = new byte[firstEntry.Length];
                         int bytesRead = entryStream.Read(uncompressedBytes);
 
-                        Assert.Equal(10, bytesRead);
-
-                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
+                        Assert.Equal(s_existingSampleData.Length, bytesRead);
+                        Assert.Equal(s_existingSampleData, uncompressedBytes);
                     }
 
                     using (Stream entryStream = secondEntry.Open())
                     {
                         byte[] uncompressedBytes = new byte[secondEntry.Length];
                         int bytesRead = entryStream.Read(uncompressedBytes);
 
-                        Assert.Equal(4, bytesRead);
-
-                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03 }, uncompressedBytes);
+                        Assert.Equal(s_sampleDataToWrite.Length, bytesRead);
+                        Assert.Equal(s_sampleDataToWrite, uncompressedBytes);
                     }
                 }
             }
         }
 
+        // This is combinatorial.
+        // "version to extract" is 0 and 20. The valid value (20) proves that updating a ZipArchive with trailing extra
+        // field data won't touch existing entries. The invalid value (0) forces ZipArchive to write the corrected header
+        // (preserving the trailing data.)
+        // "extra field data length" is 3, 4, 7, 8 and 15. This accounts for various interpretations of trailing field data.
+        // * 3: zero valid extra fields, trailing data
+        // * 4: one valid extra field (type = 0, length = 0) and no trailing data
+        // * 7: same as above, with trailing data
+        // * 8: multiple valid extra fields, all with type = 0, length = 0. No trailing data
+        // * 15: same as above, with trailing data
         [Theory]
-        [InlineData(0x00)]
-        [InlineData(0x14)]
-        public void ZipArchive_InvalidExtraFieldData(byte validVersionToExtract)
+        [InlineData(0, 3)]
+        [InlineData(20, 3)]
+        [InlineData(0, 4)]
+        [InlineData(20, 4)]
+        [InlineData(0, 7)]
+        [InlineData(20, 7)]
+        [InlineData(0, 8)]
+        [InlineData(20, 8)]
+        [InlineData(0, 15)]
+        [InlineData(20, 15)]
```

Comment:
Not too important, I can add this later, probably in my PR since I would have to write an async version of this test.

But when you combine a bunch of inline data values, instead of adding a bunch of inline datas combining everything, you can do:

```suggestion
        [MemberData(nameof(Get_ZipArchive_InvalidExtraFieldData_Data))]
```
and then above the test you add this:
```cs
public static IEnumerable<object[]> Get_ZipArchive_InvalidExtraFieldData_Data()
{
    foreach (byte validVersionToExtract in new byte[] { 0, 20 })
    {
        foreach (ushort extraFieldDataLength in new ushort[] { 3, 4, 7, 8, 15 })
        {
            yield return new object[] { validVersionToExtract, extraFieldDataLength };
        }
    }
}
```

Reply:
It's simple enough to fix here - I've just switched it to using `MemberData`.